memegen/routes/_common.py

```diff
@@ -31,11 +32,36 @@ def samples(app):
         }
 
 
-def track(app, request, title):
-    """Log the requested content."""
+def display(title, path, mimetype='image/jpeg'):
+    """Render a webpage or raw image based on request."""
+    mimetypes = request.headers.get('Accept', "").split(',')
+    browser = 'text/html' in mimetypes
```

Comment:
If a request specifies multiple accept types that include q-values, this check will fail even if `text/html` is in the Accept header:

```python
>>> accept = 'application/json; q=0.5, text/html; q=0.25'
>>> mimetypes = accept.split(',')
>>> mimetypes
['application/json; q=0.5', ' text/html; q=0.25']
>>> 'text/html' in mimetypes
False
```

Werkzeug has methods for parsing pretty much everything in the HTTP spec:

```python
>>> 'text/html' in request.accept_mimetypes
True
```

It also provides `request.accept_mimetypes.best` if you want the mimetype with the highest q-value.

In your specific case, since you want to check if the request wants HTML, you can use [`request.accept_mimetypes.accept_html`](http://werkzeug.pocoo.org/docs/0.11/datastructures/#werkzeug.datastructures.MIMEAccept.accept_html) to check if the request has an HTML content type in the Accept header.